service/src/main/java/datawave/microservice/query/QueryManagementService.java

```diff
@@ -0,0 +1,2602 @@
+                log.error("Unknown error calling create and next. {}", queryId, e);
+                qe = new QueryException(DatawaveErrorCode.QUERY_NEXT_ERROR, e, "Unknown error calling create and next. " + queryId);
+            } else {
+                qe = (QueryException) e;
+            }
+            
+            if (queryId != null && !(qe instanceof NoResultsQueryException)) {
+                getBaseQueryMetric().setError(qe);
+            }
+            
+            throw qe;
+        }
+    }
+    
+    /**
+     * Gets the next page of results for the specified query.
+     * <p>
+     * Next can only be called on a running query. <br>
+     * If configuration allows, multiple next calls may be run concurrently for a query. <br>
+     * Only the query owner can call next on the specified query.
+     *
+     * @param queryId
+     *            the query id, not null
+     * @param currentUser
+     *            the user who called this method, not null
+     * @return a base query response containing the next page of results
+     * @throws NotFoundQueryException
+     *             if the query cannot be found
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't own the query
+     * @throws BadRequestQueryException
+     *             if the query is not running
+     * @throws QueryException
+     *             if query lock acquisition fails
+     * @throws QueryException
+     *             if the next call is interrupted
+     * @throws TimeoutQueryException
+     *             if the query times out
+     * @throws NoResultsQueryException
+     *             if no query results are found
+     * @throws QueryException
+     *             if this next task is rejected by the executor
+     * @throws QueryException
+     *             if next call execution fails
+     * @throws QueryException
+     *             if query logic creation fails
+     * @throws QueryException
+     *             if there is an unknown error
+     */
+    public BaseQueryResponse next(String queryId, DatawaveUserDetails currentUser) throws QueryException {
+        log.info("Request: next from {} for {}", ProxiedEntityUtils.getShortName(currentUser.getPrimaryUser().getName()), queryId);
+        
+        try {
+            // make sure the query is valid, and the user can act on it
+            QueryStatus queryStatus = validateRequest(queryId, currentUser);
+            
+            // make sure the state is created
+            if (queryStatus.getQueryState() == CREATE) {
+                return executeNext(queryId, currentUser);
+            } else {
+                throw new BadRequestQueryException("Cannot call next on a query that is not running", HttpStatus.SC_BAD_REQUEST + "-1");
+            }
+        } catch (Exception e) {
+            // TODO: QueryException results in VoidResponse. This differs from the original RestAPI where an exception was returned in a
```

Comment:
Just a note that these TODO's are still here.

Reply:
yeah, we can probably remove some of them, but for a bunch of them i think i still have questions that need answering.

Reply:
i will take inventory of the remaining TODOs and make a decision.

Reply:
went through and handled all of the TODOs.  I either resolved them, deleted them if they were no longer applicable, or created issues for them.